src/analyze/analyze.c

```diff
@@ -1248,6 +1322,47 @@ static int dot(int argc, char *argv[], void *userdata) {
 
         printf("digraph systemd {\n");
 
+        color_after     = (char *)malloc(strlen(DOT_COLOR_AFTER) + 1);
+        color_requires  = (char *)malloc(strlen(DOT_COLOR_REQUIRES) + 1);
+        color_requisite = (char *)malloc(strlen(DOT_COLOR_REQUISITE) + 1);
+        color_wants     = (char *)malloc(strlen(DOT_COLOR_WANTS) + 1);
+        color_conflicts = (char *)malloc(strlen(DOT_COLOR_CONFLICTS) + 1);
+        color_bindsto   = (char *)malloc(strlen(DOT_COLOR_BINDSTO) + 1);
+        color_onfailure = (char *)malloc(strlen(DOT_COLOR_ONFAILURE) + 1);
+        color_triggers  = (char *)malloc(strlen(DOT_COLOR_TRIGGERS) + 1);
+        style_after     = (char *)malloc(strlen(DOT_STYLE_AFTER) + 1);
+        style_requires  = (char *)malloc(strlen(DOT_STYLE_REQUIRES) + 1);
+        style_requisite = (char *)malloc(strlen(DOT_STYLE_REQUISITE) + 1);
+        style_wants     = (char *)malloc(strlen(DOT_STYLE_WANTS) + 1);
+        style_conflicts = (char *)malloc(strlen(DOT_STYLE_CONFLICTS) + 1);
+        style_bindsto   = (char *)malloc(strlen(DOT_STYLE_BINDSTO) + 1);
+        style_onfailure = (char *)malloc(strlen(DOT_STYLE_ONFAILURE) + 1);
+        style_triggers  = (char *)malloc(strlen(DOT_STYLE_TRIGGERS) + 1);
+
+        strcpy(color_after,      DOT_COLOR_AFTER);
```

Comment:
This should use `strdup`. Also, there should be an oom check. Also, please use `_cleanup_free_`. Also, I don't think the separate defines are useful. After all, they are used in only one place. So maybe something like this:
```C
_cleanup_free_ char *color_after = strdup("green");
_cleanup_free_ char *color_requires = strdup("black");
...
if (!color_after || !color_requires || ...)
     return log_oom();
```

Reply:
Now using strdup, oom check. Removed separate defines. Not using _cleanup_free, because global now.